File: week4/이지훈/짝지어제거하기.java
import java.util.*;

public class 짝지어제거하기 {
    public int solution(String s) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            char input = s.charAt(i);

            if (stack.size() == 0) {
                stack.push(input);
                continue;
            }

            if (input == stack.peek()) stack.pop();
            else stack.push(input);
        }

        return stack.isEmpty() ? 1 : 0;
    }

}

File: week4/윤여준/짝지어 제거하기.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(string s) {
    stack<char> st;
    for(char c : s) {
        if(st.empty() || st.top() != c) st.push(c);
        else st.pop();
    }
    
    return st.empty();
}